### PR TITLE
Feat/search todo

### PR DESCRIPTION
**Pull request by mkh-wtag**
Complete search todos and filter todos

### CODE REVIEW

File: components/domActions.js [hunk 1]
```diff
@@ -27,8 +32,30 @@ export function emptyListMessage() {
...
 export const renderTodoList = () => {
   todoWrapper.innerHTML = "";
+  let filteredTodos;
 
-  todos.forEach((todo) => {
+  const searchedText = searchTodoInput.value.toLowerCase();
+
+  switch (currentFilterState) {
+    case "complete":
+      filteredTodos = todos.filter((todo) => todo.isDone === true);
```
Comment (hhu-wtag) [inline components/domActions.js]:
```suggestion
      filteredTodos = todos.filter((todo) => todo.isDone);
```

File: components/filterTodos.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { renderTodoList } from "./domActions.js";
+import { setCurrentFilterState } from "../entry.js";
+import {
+  filterButtonWrapper,
+  filterAllButton,
+  filterCompleteButton,
+  filterIncompleteButton,
+} from "./domElements.js";
+
+const filterButtons = filterButtonWrapper.getElementsByClassName("button");
```
Comment (hhu-wtag) [inline components/filterTodos.js:11]: The class name is too generic and can cause unexpected bugs. Let's use a specific name
```suggestion
const filterButtons = filterButtonWrapper.getElementsByClassName("filter-button");
```

File: components/filterTodos.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+function selectedClass(clicked) {
+  for (let current of filterButtons) {
+    current.classList.remove("active");
+  }
```
Comment (hhu-wtag) [inline components/filterTodos.js]:
```suggestion
  for (let button of filterButtons) {
    button.classList.remove("active");
  }
```

File: components/filterTodos.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  clicked.classList.add("active");
+}
+
+export function filterAll() {
+  setCurrentFilterState("all");
```
Comment (hhu-wtag) [inline components/filterTodos.js]: Let's use const here. 
```suggestion
  import { ALL, COMPLETE, INCOMPLETE } from "../const.js"
  setCurrentFilterState(ALL);
```

**Review by hhu-wtag — APPROVED** [review]
Nice work :rocket:

File: components/filterTodos.js (hunk 2 above)
Comment (mkh-wtag) [reply components/filterTodos.js:11]: It has to be generic as it contains global styles. That's why I targeted the wrapper first and then the buttons.

**Pull request merged**